internal/backend/runtime/omni/audit/state.go

```diff
@@ -14,14 +14,22 @@ import (
 
 // WrapState wraps the given state with audit log state.
 func WrapState(s state.State, l *Log) state.State {
-	return &auditState{
+	st := &auditState{
 		state:  s,
 		logger: l,
 	}
+
+	st.wrappedSelfState = state.WrapCore(st)
+
+	return st
 }
 
 type auditState struct {
-	state  state.State
+	state state.State
+
+	// we wrap this auditState itself to be able to implement Modify and ModifyWithResult methods
+	wrappedSelfState state.State
```

Comment:
This is the only meaningful code change in this PR - I did this trick to implement modify and modifywithresult, as they under the hood reuse `UpdateWithConflicts`.